### DISCUSSION TITLE
PowerOnState 4 for Shelly plus 2PM

### DISCUSSION THREAD

**Discussion by umtauscher** [arendst/Tasmota#20891] [General]
High, I have just flashed the above mentioned Shelly. I need a setting, where I have continous  power to relays 1 and normal operation on relais 2. 
Unfortunately  setting PowerOnState to 4 changes both to On.
I have read about rules, but all posted workarrounds don't work as I need it. Could anyone please help me with this?
Thanks in Advance

**Comment by sfromis** [comment]
If you want relay 1 to be on 100% of the time, never off, you could change the gpio definition to `Output Hi` instead. This means that Tasmota does not even "see" a relay, and the other relay should have number 1 instead.

**Comment by umtauscher** [comment]
Thanks, I Will try that. What about the energy measurement. Does this still work?

**Reply by sfromis** [reply]
I'd expect so. Probably use `SetOption21` to keep the energy monitoring going while the "normal" relay is off.

**Comment by brjhaverkamp** [comment]
Hi all, 
I found this discussion as I have the same issue. I use a Shelly 2.5PM. Relay one is used to measure the power usage towards my EV. There is no button attached.  The second relay is used  to switch the light in the carport. 
I noticed that the relay to the EV sometime switches. I assume ghost switches. So I want to fix that relay to always on. The carports lights need to stay switchable, of course.

I'm going to try the above tip of umtauscher to change the gpio  definition.
But wouldn't in make the most sense to have Poweronstate<x>, in line with the syntax of many other commands.
That way I could fix relay 1 to ON, and leave relay 2 switchable 
Regards,

Bert

**Comment by arendst** [comment]
With latest version you might want to try command ``PowerLock1 1`` after you enabled power for relay1 (``power1 1``).

Powerlock disables power control and should keep relay 1 to the last state set with ``power1 1``.

**Comment by brjhaverkamp** [comment]
Hi, thanks, that is indeed a confirmation I needed. 
I found that command and tried it. But unfortunately, the switch did somehow still change to `off`, and then be locked in the `off` state. Which was very surprising to me. Could this be a bug? I can test some more if needed. 
BTW, From the documentation it wasn't clear to me if powerlock<x> will also work through reboots, e.g.  after power failures. Does it?

**Reply by sfromis** [reply]
When it comes to those configuration commands, the expectation should be that they are persistent, saved across reboots (unless you've taken steps to turn off automatic saving). And yes, this includes `PowerLock` settings per channel.

Note that using `PowerLock` does mean that current power state will always stick at current state, no matter what. You can still have other features influencing state, like `PowerOnState`, or if your power state was not saved (as already mentioned). In other words, `PowerLock` will block attempts at *changing* power state (typically the `Power` command`), but will not influence how power state is handled during boot.

**Comment by brjhaverkamp** [comment]
Hi all, I have used Powerlock1 1 now for a while and monitored it. 
I thought it was stable after no problems for almost a week, and then the outlet switched off again.. 
It is now in the off state, and still locked. So I have to unlock it first, switch it on and then lock it again.
What could this be?

**Reply by sfromis** [reply]
I already gave you some suggestion about that nearly 3 weeks ago.

**Reply by brjhaverkamp** [reply]
Hi sfromis, 
Appologies. I missed that one. But also I am not sure what you suggest. 
I can't use powerOnState, as it affects both relays in the shelly 2.5. And the EV part should be on by default, while the Carport light should be off by default.

Also I don't understand what has changed the power state in my case. I haven't used PowerOnState commands. 
And my uptime is 19T16:21:42. That's 19 Days, right? So it wasn't a restart/reboot either. 
Btw, How does the device react to input on the physical switching side? I haven't connected anything, so I voltage is floating, and noise might trigger it.

What would you advice?

Regards,

Bert

**Reply by sfromis** [reply]
The point was not to *use* `PowerOnState` (technically, you are *always* using the setting, even if you have not set it yourself), but it will always affect how the relays are handled. `PowerLock` is not overriding `PowerOnState`, but *only* blocking you from changing relay state during normal operation.

Assuming default `PowerOnState 3`, it matters a lot what the saved power state is, what you could be doing to disable saving, or if it as even saved as expected. You could make sure to have the power on, and then do a normal restart of Tasmota, to ensure that you have the desired value saved.

My experience with a simple plug is that `PowerLock` also blocks default actions for Button inputs. Again, the key is "during normal operation".

Anyway, my initial suggestion in this thread is much simpler, and outside of any actions to influence (or even see) power state.

**Comment by brjhaverkamp** [comment]
Hi sfromis,

I followed your initial advice and adjusted the template. Interesting. Never touched that part of Tasmota:-). 
So thanks for that.

Still, I am wondering what triggered the switching. 
The device hadn't restarted. (uptime 19 days), I hadn't changed parameters. And it couldn't have been ghost switches, because of the Powerlock.

So in all: Thanks for the support. much appreciated. 
I can now charge my car again without nasty suprises of a still empty battery in the morning!

Bert

**Comment by brjhaverkamp** [comment]
Ok, I went the template way. I set GPIO4 to Output Hi. Sofar the charger is always on. And I can control the light with the one remaining button in the web view. Which is kind of nice. No bogus toggle that doesn't work. 

However, I noticed that the power measurement on my EV charger _only_ reports power when I have the light switch turned on?!?
If I switch the light off, both reported powers on O1 and O2 drop to zero.  The charger still draws power, but it is just not measured/reported anymore.

This was an unexpected side-effect  And since it is getting dark here now in the evening, I didn't even find out initially, since I had the light on while testing:-)  Is this explainable from how the firmware works? And is there a way to solve this?
I added two screenshots of the on versus off situation.
![Screenshot from 2024-10-05 08-06-03](https://github.com/user-attachments/assets/51446215-c5f3-466d-972e-39faba0ecc81)
![Screenshot from 2024-10-05 08-05-45](https://github.com/user-attachments/assets/97d16d3e-c12e-4e4c-8ba2-ca6ef069a1ad)

Regards,

Bert

**Reply by barbudor** [reply]
`SetOption21 1`
https://tasmota.github.io/docs/Commands/#setoption21

**Reply by sfromis** [reply]
That detail was already suggested in the start of this thread.

**Comment by brjhaverkamp** [comment]
Thanks barbudor.
And appologis sfromos, Missed that one.. not making a good impression here ..;-)
